### PR TITLE
logger: Make CC_DESC string length indivisible by four

### PR DESCRIPTION
**Pull request by ktrzcinx**
This patch should fix logger without ABI change.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

### CODE REVIEW

**Comment by paulstelian97** [comment]
Why do we want it to not be divisible by 4? If it is divisible it would lack a NUL terminator or something? (or have an off-by-one calculation of size)

**Comment by jajanusz** [comment]
W/A for issue that should be finally solved by https://github.com/thesofproject/sof/pull/2522, but is taking too long to get merged and makes logger unusable in CI for weeks.

**Comment by jajanusz** [comment]
Now I'm thinking if ranlib may be messing with it (it uses one from gcc instead of xtensa), can you try to repro issue with this change:
```diff
diff --git a/scripts/cmake/xtensa-toolchain.cmake b/scripts/cmake/xtensa-toolchain.cmake
index c65caa2a9..bf2d6b362 100644
--- a/scripts/cmake/xtensa-toolchain.cmake
+++ b/scripts/cmake/xtensa-toolchain.cmake
@@ -56,6 +56,7 @@ endif()

 find_program(CMAKE_LD NAMES "${CROSS_COMPILE}ld" PATHS ENV PATH NO_DEFAULT_PATH)
 find_program(CMAKE_AR NAMES "${CROSS_COMPILE}ar" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_RANLIB NAMES "${CROSS_COMPILE}ranlib" PATHS ENV PATH NO_DEFAULT_PATH)
 find_program(CMAKE_OBJCOPY NAMES "${CROSS_COMPILE}objcopy" PATHS ENV PATH NO_DEFAULT_PATH)
 find_program(CMAKE_OBJDUMP NAMES "${CROSS_COMPILE}objdump" PATHS ENV PATH NO_DEFAULT_PATH)
```

File: scripts/cmake/xtensa-toolchain.cmake [hunk 1]
```diff
@@ -56,6 +56,7 @@ endif()
 
 find_program(CMAKE_LD NAMES "${CROSS_COMPILE}ld" PATHS ENV PATH NO_DEFAULT_PATH)
```
Comment (dbaluta) [inline scripts/cmake/xtensa-toolchain.cmake]: @ktrzcinx I'm totally puzzled by this PR.

Please explain in the commit message what is the ABI change that we are fixing. Also explain WHY this change is needed.

Comment (ktrzcinx) [reply scripts/cmake/xtensa-toolchain.cmake]: changed

**Comment by mmaka1** [comment]
Fixes #2507

File: scripts/cmake/xtensa-toolchain.cmake (hunk 1 above)
Comment (marc-hb) [reply scripts/cmake/xtensa-toolchain.cmake]: I think a few words in the code itself wouldn't hurt either, for now it's just pure maths :-)

**Comment by marc-hb** [comment]
Does anyone know why the exact same `check-sof-logger` test row still fails with a different error? Just curious.

**Comment by ktrzcinx** [comment]
> Does anyone know why the exact same check-sof-logger test row still fails with a different error? Just curious.

It's a good question, @xiulipan do you know the root cause? Locally I can create output logs without any problem but test failed.

**Comment by xiulipan** [comment]
@ktrzcinx No, I can only figure out there is indeed some limitation when using flex structure in ELF files. The best solution is to avoid using this kind of structure in ELF sections.

**Comment by lgirdwood** [comment]
Jenkins known issues.
